components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAO.java

```diff
@@ -411,8 +411,25 @@ private void storeAccessToken(String accessToken, String consumerKey, AccessToke
                     throw new IdentityOAuth2Exception(errorMsg, e);
                 }
 
-                recoverFromConAppKeyConstraintViolation(accessToken, consumerKey, accessTokenDO,
-                        connection, userStoreDomain, retryAttempt + 1);
+                boolean isPostgreSQL = false;
+                Connection pgConnection = IdentityDatabaseUtil.getDBConnection();
+                try {
+                    if (StringUtils.containsIgnoreCase(pgConnection.getMetaData().getDriverName(), "PostgreSQL")) {
+                        isPostgreSQL = true;
+                        recoverFromConAppKeyConstraintViolation(accessToken, consumerKey, accessTokenDO,
+                                pgConnection, userStoreDomain, retryAttempt + 1);
+                    }
+                } catch (SQLException ex) {
+                    log.error("Error while retrieving the connection metadata", ex);
```

Comment:
As we don't throw here the issue will be already there with postgreSQL , if there is a connection issue. But if we throw fro here it will break the flow for other db types as well. So having only a log here is fine, but lets extract the above logic to a private one.